wordle-web/src/views/EditorView.vue

```diff
@@ -0,0 +1,164 @@
+    <h3>{{ deleteError }}</h3>
+  </v-container>
+</template>
+
+<script setup lang="ts">
+import { EditorService } from '@/scripts/editorService'
+import Axios from 'axios'
+import { onMounted, ref } from 'vue'
+
+const allWords = ref(['', ''])
+const allIsCommon = ref(['', ''])
+
+const addError = ref('')
+const add = ref('')
+const checkbox1 = ref(false)
+const updateError = ref('')
+const update = ref('')
+const checkbox2 = ref(false)
+const deleteError = ref('')
+const remove = ref('')
+
+onMounted(async () => {
+  allWords.value = await EditorService.getAllWords()
+  allIsCommon.value = await EditorService.getAllIsCommon()
+})
+
+async function addWord() {
+  let newWord = add.value.toLocaleLowerCase()
+  if (newWord.length != 5) {
+    addError.value = 'Word length must be 5 characters.'
+  } else if (allWords.value.includes(newWord)) {
+    addError.value = 'Word already exists.'
+  } else {
+    Axios.post('/Word/AddWord?word=' + newWord + '&isCommon=' + checkbox1.value)
+      .then(function (response) {
+        console.log(response)
+      })
+      .catch(function (error) {
+        console.log(error)
+      })
+    addError.value = 'New Word Added: ' + newWord + ', Is Common: ' + checkbox1.value
+    allWords.value = await EditorService.getAllWords()
+    allIsCommon.value = await EditorService.getAllIsCommon()
+  }
+}
+
+async function updateWord() {
+  let newWord = update.value.toLocaleLowerCase()
+  if (newWord.length != 5) {
+    updateError.value = 'Word length must be 5 characters.'
+  } else if (!allWords.value.includes(newWord)) {
+    updateError.value = 'Word does not exist.'
+  } else {
+    Axios.post('/Word/UpdateWord?word=' + newWord + '&isCommon=' + checkbox2.value)
+      .then(function (response) {
+        console.log(response)
+      })
+      .catch(function (error) {
+        console.log(error)
+      })
+    updateError.value = 'Updated Word: ' + newWord + ', Is Common: ' + checkbox2.value
+    allWords.value = await EditorService.getAllWords()
+    allIsCommon.value = await EditorService.getAllIsCommon()
+  }
+}
+
+async function removeWord() {
+  let newWord = remove.value.toLocaleLowerCase()
+  if (newWord.length != 5) {
+    deleteError.value = 'Word length must be 5 characters.'
+  } else if (!allWords.value.includes(newWord)) {
+    deleteError.value = 'Word does not exist.'
+  } else {
+    Axios.post('/Word/DeleteWord?word=' + newWord)
+      .then(function (response) {
+        console.log(response)
+      })
+      .catch(function (error) {
+        console.log(error)
+      })
+    deleteError.value = 'Removed Word: ' + newWord
+    allWords.value = await EditorService.getAllWords()
+    allIsCommon.value = await EditorService.getAllIsCommon()
+  }
+}
+
+function validWords(word: string): string[] {
+  if (word !== '') {
+    return allWords.value.filter((element) => {
+      return element.substring(0, word.length) == word
+    })
+  } else {
+    return new Array<string>()
+  }
+}
```

Comment:
This should also be on the backend. Lots of lag, it should work better if you set up a GET request and page the data.

Reply:
Noted! I'll look into implementing that!